src/Symfony/Component/Finder/Iterator/VcsIgnoredFilterIterator.php

```diff
@@ -36,11 +36,12 @@ final class VcsIgnoredFilterIterator extends \FilterIterator
     public function __construct(\Iterator $iterator, string $baseDir)
     {
         $this->baseDir = $this->normalizePath($baseDir);
-
-        foreach ($this->parentDirectoriesUpwards($this->baseDir) as $parentDirectory) {
-            if (@is_dir("{$parentDirectory}/.git")) {
-                $this->baseDir = $parentDirectory;
-                break;
+        if (!@is_dir("{$this->baseDir}/.git")) {
+            foreach ($this->parentDirectoriesUpwards($this->baseDir) as $parentDirectory) {
+                if (@is_dir("{$parentDirectory}/.git")) {
+                    $this->baseDir = $parentDirectory;
+                    break;
+                }
             }
         }
```

Comment:
Maybe we can add the base directory to the `foreach` iteration to avoid duplicating the check?
```suggestion
        foreach ([$this->baseDir, ...$this->parentDirectoriesUpwards($this->baseDir)] as $directory) {
            if (@is_dir("{$directory}/.git")) {
                $this->baseDir = $directory;
                break;
            }
        }
```